src/Query/Condition.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\entity\Query;
+
+/**
+ * Represents a single access condition.
+ */
+final class Condition {
```

Comment:
A quick search of Drupal core shows that the `final` keyword is only used for Event objects. Although, I can see why it would be useful for this value object.

We have this same class, almost exactly for JSON API and for the work done for [Issue #777578](https://www.drupal.org/project/drupal/issues/777578), [here](https://cgit.drupalcode.org/entity_access_policies/tree/src/Condition/Value?h=8.x-1.x). It definitely seems like this is a needed concept.

Reply:
I tend to stick "final" to every class whose usage is hardcoded. Conditions can't be swapped out, so there's no point in allowing them to be extended.

That said, I'm not married to this detail. If it doesn't pass core review, we'll remove it at that point.

Reply:
For me sounds like a good idea to make this final. If there comes up a usecase removing it is always possible

Reply:
One reason final isn't that much a thing in core is that you can't mock these classes anymore. On the other hand, Condition for example is a clear value object.